mirgecom/boundary.py

```diff
@@ -38,91 +40,264 @@
-
-    def _dummy_temperature_func(self, temperature_minus, **kwargs):
-        return -temperature_minus
+            self._bnd_state_func = self._identical_state
 
-    def _dummy_state_func(self, state_minus, **kwargs):
-        return state_minus
+        if not self._bnd_temperature_func:
+            self._bnd_temperature_func = self._temperature_for_prescribed_state
+        if not self._grad_num_flux_func:
+            self._grad_num_flux_func = gradient_flux_central
+
+        if not self._cv_gradient_flux_func:
+            self._cv_gradient_flux_func = self._gradient_flux_for_prescribed_cv
+        if not self._temperature_grad_flux_func:
+            self._temperature_grad_flux_func = \
+                self._gradient_flux_for_prescribed_temperature
+
+        if not self._viscous_flux_func:
+            self._viscous_flux_func = self._viscous_flux_for_prescribed_state
+        if not self._bnd_grad_cv_func:
+            self._bnd_grad_cv_func = self._identical_grad_cv
+        if not self._bnd_grad_temperature_func:
+            self._bnd_grad_temperature_func = self._identical_grad_temperature
 
     def _boundary_quantity(self, discr, btag, quantity, **kwargs):
         """Get a boundary quantity on local boundary, or projected to "all_faces"."""
+        from grudge.dof_desc import as_dofdesc
+        btag = as_dofdesc(btag)
         if "local" in kwargs:
             if kwargs["local"]:
                 return quantity
         return discr.project(btag, btag.with_dtag("all_faces"), quantity)
 
-    def inviscid_divergence_flux(self, discr, btag, gas_model, state_minus,
-                                 **kwargs):
+    def _boundary_state_pair(self, discr, btag, gas_model, state_minus, **kwargs):
+        return TracePair(btag,
+                         interior=state_minus,
+                         exterior=self._bnd_state_func(discr=discr, btag=btag,
+                                                       gas_model=gas_model,
+                                                       state_minus=state_minus,
+                                                       **kwargs))
+
+    def _temperature_for_prescribed_state(self, discr, btag,
+                                          gas_model, state_minus, **kwargs):
+        boundary_state = self._bnd_state_func(discr=discr, btag=btag,
+                                              gas_model=gas_model,
+                                              state_minus=state_minus,
+                                              **kwargs)
+        return boundary_state.temperature
+
+    def _identical_state(self, state_minus, **kwargs):
+        return state_minus
+
+    def _identical_grad_cv(self, grad_cv_minus, **kwargs):
+        return grad_cv_minus
+
+    def _identical_grad_temperature(self, grad_t_minus, **kwargs):
+        return grad_t_minus
+
+    def _gradient_flux_for_prescribed_cv(self, discr, btag, gas_model, state_minus,
```

Comment:
What type is `state_minus`?

Reply:
([8344799](https://github.com/illinois-ceesd/mirgecom/pull/581/commits/8344799ab04e877403b0e5162d2f6f9ece3505b4))